src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -203,10 +206,13 @@ public String url() {
      *  Note: to avoid deadlocks, when nesting locks we ALWAYS need to lock on the guard first, THEN the WorkflowRun.
      *  Synchronizing this helps ensure that fields are not mutated during a {@link #save()} operation, since that locks on the Run.
      */
-    private synchronized Object getMetadataGuard() {
-        if (metadataGuard == null) {
-            metadataGuard = new Object();
+    private Object getMetadataGuard() {
+        synchronized (this) {
+            if (metadataGuard == null) {
+                metadataGuard = new Object();
+            }
         }
```

Comment:
Did you mean to use double-checked locking (with `volatile` making it safer)?

```suggestion
        if (metadataGuard == null) {
            synchronized (this) {
                if (metadataGuard == null) {
                    metadataGuard = new Object();
                }
            }
        }
```

Reply:
Oh, good point, I can do that. Really all I wanted to do was be able to add the assertion outside of the scope of the `synchronized` block. Not needing to synchronize on `this` at all when `metadataGuard` is initialized is a nice bonus though.

Reply:
Done in 0c1fde4.